src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitiesSql.java

```diff
@@ -0,0 +1,364 @@
+    }
+
+    private Map<String, teammates.storage.sqlentity.Section> migrateSectionChain(
+            teammates.storage.sqlentity.Course newCourse) {
+        List<CourseStudent> oldStudents = ofy().load().type(CourseStudent.class).filter("courseId", newCourse.getId())
+                .list();
+        Map<String, teammates.storage.sqlentity.Section> sections = new HashMap<>();
+        Map<String, List<CourseStudent>> sectionToStuMap = oldStudents.stream()
+                .collect(Collectors.groupingBy(CourseStudent::getSectionName));
+
+        for (Map.Entry<String, List<CourseStudent>> entry : sectionToStuMap.entrySet()) {
+            String sectionName = entry.getKey();
+            List<CourseStudent> stuList = entry.getValue();
+            teammates.storage.sqlentity.Section newSection = createSection(newCourse, sectionName);
+            sections.put(sectionName, newSection);
+            migrateTeams(newCourse, newSection, stuList);
+            saveEntityDeferred(newSection);
+        }
+        return sections;
+    }
+
+    private void migrateTeams(teammates.storage.sqlentity.Course newCourse,
+            teammates.storage.sqlentity.Section newSection, List<CourseStudent> studentsInSection) {
+        Map<String, List<CourseStudent>> teamNameToStuMap = studentsInSection.stream()
+                .collect(Collectors.groupingBy(CourseStudent::getTeamName));
+        for (Map.Entry<String, List<CourseStudent>> entry : teamNameToStuMap.entrySet()) {
+            String teamName = entry.getKey();
+            List<CourseStudent> stuList = entry.getValue();
+            teammates.storage.sqlentity.Team newTeam = createTeam(newSection, teamName);
+            migrateStudents(newCourse, newTeam, stuList);
+            saveEntityDeferred(newTeam);
+        }
+    }
+
+    private void migrateStudents(teammates.storage.sqlentity.Course newCourse, teammates.storage.sqlentity.Team newTeam,
+            List<CourseStudent> studentsInTeam) {
+        for (CourseStudent oldStudent : studentsInTeam) {
+            teammates.storage.sqlentity.Student newStudent = createStudent(newCourse, newTeam, oldStudent);
+            saveEntityDeferred(newStudent);
+        }
+    }
+
+    private teammates.storage.sqlentity.Course createCourse(Course oldCourse) {
+        teammates.storage.sqlentity.Course newCourse = new teammates.storage.sqlentity.Course(
+                oldCourse.getUniqueId(),
+                oldCourse.getName(),
+                oldCourse.getTimeZone(),
+                oldCourse.getInstitute());
+        newCourse.setDeletedAt(oldCourse.getDeletedAt());
+        newCourse.setCreatedAt(oldCourse.getCreatedAt());
+
+        saveEntityDeferred(newCourse);
+        return newCourse;
+    }
+
+    private teammates.storage.sqlentity.Section createSection(teammates.storage.sqlentity.Course newCourse,
+            String sectionName) {
+        // if (sectionName.equals(Const.DEFAULT_SECTION)) {
```

Comment:
@FergusMok is there any update on this?

Reply:
I believe its being fixed here https://github.com/TEAMMATES/teammates/pull/13040